### PR TITLE
Chat page refactoring

### PR DESCRIPTION
**Pull request by philipperolet**
OPTIONAL Small refactoring of the chat index page that I had to do to prepare for the new chat flow using Openai fns

IMHO can be pushed anyways, wdyt?
On Yes / no will either merge / deploy or close the PR

### CODE REVIEW

File: front/pages/w/[wId]/u/chat/[cId]/index.tsx [hunk 1]
```diff
@@ -679,207 +679,212 @@ export default function AppChat({
...
+    if (res.isErr()) {
+      m.push({
+        role: "error",
+        message: res.error.message,
+      } as ChatMessageType);
+      // console.log("ERROR", res.error);
+      setMessages(m);
+      setResponse(null);
+      setLoading(false);
+      return;
```
Comment (spolu) [inline front/pages/w/[wId]/u/chat/[cId]/index.tsx:722]: beware that return would skip the runAssistant if retrieval had failed. You might runRetrieval now right?

File: front/pages/w/[wId]/u/chat/[cId]/index.tsx [hunk 2]
```diff
@@ -679,207 +679,212 @@ export default function AppChat({
...
         m.push({
           role: "error",
-          message: res.error.message,
+          message: event.content.message,
         } as ChatMessageType);
-        // console.log("ERROR", res.error);
         setMessages(m);
         setResponse(null);
         setLoading(false);
         return;
```
Comment (spolu) [inline front/pages/w/[wId]/u/chat/[cId]/index.tsx:738]: same here

File: front/pages/w/[wId]/u/chat/[cId]/index.tsx [hunk 3]
```diff
@@ -679,207 +679,212 @@ export default function AppChat({
...
+        if (event.content.block_name === "DATASOURCE") {
+          if (e.error) {
+            m.push({
+              role: "error",
+              message: e.error,
+            } as ChatMessageType);
+            setMessages(m);
+            setResponse(null);
+            setLoading(false);
+            return;
```
Comment (spolu) [inline front/pages/w/[wId]/u/chat/[cId]/index.tsx:751]: same here

**Review by spolu — COMMENTED** [review]
Looks great!

Only problem is the case of failure of runRetrieval that will let runAssistant run erroneously in the current state of affair.

**Comment by philipperolet** [comment]
did it; all errors seemed handlable with the same code in a try catch block
added 2 error cases that weren't there -- unsure about that if you can check the comms

File: front/pages/w/[wId]/u/chat/[cId]/index.tsx [hunk 4]
```diff
@@ -679,13 +679,125 @@ export default function AppChat({
...
+      if (event.type === "block_execution") {
+        const e = event.content.execution[0][0];
+        if (event.content.block_name === "MODEL" && e.error)
+          throw new Error(e.error);
+        if (event.content.block_name === "OUTPUT") {
+          if (!e.error) {
+            m.push(e.value);
+            setMessages(m);
+            setResponse(null);
+          } else throw new Error("Error in chat assistant execution.");
```
Comment (philipperolet) [inline front/pages/w/[wId]/u/chat/[cId]/index.tsx:776]: here previous code ignored the case e.error true

**Review by spolu — DISMISSED** [review]
LGTM

**Pull request merged**